inc/policyremoveapplication.class.php

```diff
@@ -90,13 +91,58 @@ public function unicityCheck($value, $itemtype, $itemId, PluginFlyvemdmFleet $fl
     * @return array|bool
     */
    public function getMqttMessage($value, $itemtype, $itemId) {
-      if (! $this->integrityCheck($value, $itemtype, $itemId)) {
+      if (!$this->integrityCheck($value, $itemtype, $itemId)) {
          return false;
       }
 
       $array = [
-            $this->symbol  => $value
+         $this->symbol  => $value,
       ];
       return $array;
    }
+
+   /**
+    * @param PluginFlyvemdmFleet $fleet
+    * @param mixed $value
+    * @param mixed $itemtype
+    * @param int $itemId
+    * @return bool
+    */
+   public function apply(PluginFlyvemdmFleet $fleet, $value, $itemtype, $itemId) {
+      $this->canApply = false;
+
+      // force the itemtype for packages
+      if (!($itemtype instanceof PluginFlyvemdmPackage)) {
+         $itemtype = new PluginFlyvemdmPackage();
+      }
+
+      // the itemId wasn't send and the itemtype is new
+      if ($itemtype->isNewItem() && !$itemId) {
+         Session::addMessageAfterRedirect(__('An application ID is required', 'flyvemdm'),
+            false, ERROR);
+         return false;
+      }
+
+      // the itemId is send but the itemtype is new, lest's try to load the info of the package
+      if ($itemtype->isNewItem() && $itemId) {
+         if (!$itemtype->getFromDB($itemId)) {
+            Session::addMessageAfterRedirect(__('The application does not exists', 'flyvemdm'),
+               false, ERROR);
+            return false;
+         }
+      }
+
+      // the package is loaded, let's create the task
+      $task = new PluginFlyvemdmTask();
+      if (!$task->add([
+         'plugin_flyvemdm_fleets_id'   => $fleet->getID(),
+         'plugin_flyvemdm_policies_id' => $this->getPolicyData()->getID(),
+         'value'                       => $itemtype->getField('package_name'),
```

Comment:
use  the value received as argument here.

Also, for readability, I think it is better to call $task->add()  out of the if () condition. You can write 
```php
if ($task->isNewItem()) {
   return false;
}